tests/test_version.py

```diff
@@ -1,7 +1,42 @@
 from __future__ import annotations
 
+from _pytest.monkeypatch import MonkeyPatch
+
+from tox import __version__
+from tox.plugin.manager import MANAGER
+from tox.pytest import ToxProjectCreator
+
 
 def test_version() -> None:
     from tox import __version__
 
     assert __version__
+
+
+def test_version_without_plugin(tox_project: ToxProjectCreator) -> None:
+    outcome = tox_project({"tox.ini": ""}).run("--version")
+    outcome.assert_success()
+    assert __version__ in outcome.out
+    assert "plugin" not in outcome.out
+
+
+def test_version_with_plugin(tox_project: ToxProjectCreator, monkeypatch: MonkeyPatch) -> None:
+    def dummy_plugin():
+        class MockModule:
+            __file__ = "dummy-path"
+
+        class MockEggInfo:
+            project_name = "dummy-project"
+            version = "1.0"
+
+        return [(MockModule, MockEggInfo)]
+
+    monkeypatch.setattr(MANAGER.manager, "list_plugin_distinfo", dummy_plugin)
```

Comment:
@gaborbernat I dug deep into the different plugin tests but I was not able to create a `tox_project` with registered plugins that would get displayed in `--version`. Maybe I'm missing something with how to properly use `tox_project` and `register_inline_plugin` but the manager could not find them.

Instead I looked at how it was done in `tox3` like @jugmac00 suggested, and used `MonkeyPatch` to get a plugin as a response from the plugin manager.